train.py

```diff
@@ -0,0 +1,122 @@
+import os
+import time
+
+import hydra
+import torch
+import psutil
+import imageio
+import numpy as np
+from omegaconf import DictConfig
+from torchvision.utils import save_image
+
+import wandb
+from utils import get_interpolations
+from data_loader import get_data_loaders
+from models.autoencoder import Autoencoder
+from train_utils import test_epoch, train_epoch, prepare_loss_function
+
+
+def initialize_model(cfg: DictConfig, train_loader, test_loader, input_size):
+    if cfg.model.type == "AE":
+        model = Autoencoder(cfg, train_loader=train_loader, test_loader=test_loader, input_size=input_size)
+    else:
+        raise NotImplementedError("Model type not implemented")
+    return model
+
+
+def train(cfg: DictConfig, autoencoder: Autoencoder, loss_function: torch.nn.Module):
+    os.makedirs(cfg.logger.results_path, exist_ok=True)
+    os.makedirs(cfg.logger.checkpoint_path, exist_ok=True)
+
+    torch.manual_seed(cfg.system.seed)
+
+    train_autoencoder(cfg, autoencoder, loss_function)
+    draw_interpolation_grid(cfg, autoencoder)
+
+
+def train_autoencoder(cfg: DictConfig, autoencoder: Autoencoder, loss_function: torch.nn.Module):
+    total_start_time = time.time()
+    process = psutil.Process()
+
+    optimizer = torch.optim.Adam(autoencoder.model.parameters(), lr=cfg.train.lr)
+    for epoch in range(1, cfg.train.epochs + 1):
+        train_epoch(
+            autoencoder, autoencoder.train_loader, optimizer, autoencoder.device, cfg.train.log_interval, epoch, loss_function
+        )
```

Comment:
```suggestion
        train_epoch(
            autoencoder=autoencoder,
            loader=autoencoder.train_loader,
            optimizer=optimizer,
            device=autoencoder.device,
            log_interval=cfg.train.log_interval,
            epoch=epoch,
            loss_fn=loss_function,
        )
```

Let's break code like this into multiple lines, that way it's easier for me to read :) 

You should use [magic trailing commas](https://black.readthedocs.io/en/stable/the_black_code_style/current_style.html#the-magic-trailing-comma) to let `black` know that you want this split.

Also please use named parameters whenever there are multiple non-idiomatic arguments, again for readability gains